workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs

```diff
@@ -42,17 +40,18 @@ protected override void OnUpdate()
 
         private void OnAddEntity(AddEntityOp op)
         {
-            view.CreateEntity(op.EntityId.Id);
+            worker.CreateEntity(op.EntityId.Id);
         }
 
         private void OnRemoveEntity(RemoveEntityOp op)
         {
-            view.RemoveEntity(op.EntityId.Id);
+            worker.RemoveEntity(op.EntityId.Id);
         }
 
         private void OnDisconnect(DisconnectOp op)
         {
-            view.Disconnect(op.Reason);
+            EntityManager.RemoveComponent<IsConnected>(worker.WorkerEntity);
+            EntityManager.AddSharedComponentData(worker.WorkerEntity, new OnDisconnected { ReasonForDisconnect = op.Reason });
```

Comment:
Why is this a shared component?

Reply:
this was always a shared component. simply moved the code from view.disconnect to here.

Though I agree that a shared component data is probably not needed here.I think this was done by @samiwh ?